contracts/contracts/strategies/ConvexEthMetaStrategy.sol

```diff
@@ -257,19 +337,169 @@ contract ConvexEthMetaStrategy is InitializableAbstractStrategy {
         );
 
         // Burn all OETH
-        uint256 oethBalance = oeth.balanceOf(address(this));
-        IVault(vaultAddress).burnForStrategy(oethBalance);
+        uint256 oethToBurn = oeth.balanceOf(address(this));
+        IVault(vaultAddress).burnForStrategy(oethToBurn);
+
+        // Get the strategy contract's ether balance.
+        // This includes all that was removed from the Metapool and
+        // any ether that was sitting in the strategy contract before the removal.
+        uint256 ethBalance = address(this).balance;
+        // Convert all the strategy contract's ether to WETH and transfer to the vault.
+        weth.deposit{ value: ethBalance }();
+        require(
+            weth.transfer(vaultAddress, ethBalance),
+            "Transfer of WETH not successful"
+        );
+
+        emit Withdrawal(address(weth), address(lpToken), ethBalance);
+        emit Withdrawal(address(oeth), address(lpToken), oethToBurn);
+    }
+
+    /***************************************
+            Metapool Rebalancing
+    ****************************************/
+
+    /**
+     * @notice One-sided remove of OTokens from the Metapool which are then burned.
+     * This is used when the Metapool has too many OTokens and not enough ETH.
+     * The amount of assets in the vault is unchanged.
+     * The total supply of OTokens is reduced.
+     * The asset value of the strategy and vault is reduced.
+     * @param _lpTokens The amount of Metapool LP tokens to be burned for OTokens.
+     */
+    function removeAndBurnOTokens(uint256 _lpTokens)
+        external
+        onlyStrategist
+        improveMetapoolBalance
+    {
+        // Withdraw Metapool LP tokens from Convex pool
+        _lpWithdraw(_lpTokens);
+
+        // Convert Metapool LP tokens to ETH value
+        uint256 valueInEth = _lpTokens.mulTruncate(
+            curvePool.get_virtual_price()
+        );
+        // Apply slippage to ETH value
+        uint256 minOethAmount = valueInEth.mulTruncate(
+            uint256(1e18) - MAX_SLIPPAGE
+        );
+
+        // Remove just the OTokens from the Metapool
+        uint256 oethToBurn = curvePool.remove_liquidity_one_coin(
+            _lpTokens,
+            int128(oethCoinIndex),
+            minOethAmount,
+            address(this)
+        );
+
+        // The vault burns the OTokens from this strategy
+        IVault(vaultAddress).burnForStrategy(oethToBurn);
+
+        emit Withdrawal(address(oeth), address(lpToken), oethToBurn);
+    }
 
-        // Send all ETH and WETH on the contract, including extra
-        weth.deposit{ value: address(this).balance }();
+    /**
+     * @notice Mint OTokens and one-sided add to the Metapool.
+     * This is used when the Metapool has not enough OTokens and too many ETH.
```

Comment:
nit: I might be off here but would it sound better like this maybe: 
```
This is used when the Metapool doesn't have enough OTokens and too much ETH
```

Reply:
That's better. Thanks